homeassistant/components/knx/config_flow.py

```diff
@@ -247,6 +246,8 @@ async def async_step_tunnel(
                 user_password=None,
                 tunnel_endpoint_ia=None,
             )
+            if connection_type == CONF_KNX_TUNNELING_TCP:
```

Comment:
It's not that advanced imho. Every IP interface has some (4-9) tunnel endpoints - each have their own KNX address. Setting this address as constant has some major upsides:
- KNX DataSecure (application layer encryption) can be used without having to assign the used addresses to all tunnel endpoints - so this saves lots of drag&drop in the provisioning software (ETS)
- knowing the address makes debugging easier as the tunnel endpoint can be named and seen in ETS Group monitor or other diagnostic tools
- Address conflicts with other software using the IP interface can be avoided. These may result in unsuccessful connections or sometimes maybe even in disconnects of the first connected tunnel. (Eg. if HA connects to the first and ETS is set up to use the same tunnel).

Reply:
The config flow for a TCP tunneling connection is currently a 2-3 step flow:
1. select Auto / Tunnel / Routing
Auto is a single step - no further steps.
<img width="613" alt="Bildschirmfoto 2024-12-01 um 20 06 38" src="https://github.com/user-attachments/assets/4b535de1-99a0-4aec-87e3-94bc9ede8ee8">

2. select the IP interface - this step is skipped if only one interface is found (which is probably the case for most installations)
<img width="485" alt="Bildschirmfoto 2024-12-01 um 20 06 52" src="https://github.com/user-attachments/assets/e7b17f8f-f80f-47b7-8dea-091226a341df">

3. select the tunnel endpoint address - this step is skipped for non-TCP capable interfaces (newer ones) as the selection feature is not supported for those.
<img width="431" alt="Bildschirmfoto 2024-12-01 um 20 07 01" src="https://github.com/user-attachments/assets/e3a7c347-a529-4463-a2fd-3a02da695d70">

Reply:
To give a bit more details, since every KNX installation is very unique in its mix of devices and configuration, I would argue that there is no "simple" use case at all. Especially when newer installations will move to/use KNX IP Secure and Data Secure.

Reply:
Right. For really simple single-line, single-interface, non-secure installations... there is always "Automatic" - which is a single step flow.

Reply:
But like, what option would the user choose? Because I know quite a bunch of (older) people who have a smart home, but don't know they run KNX. So when I take those people in mind, they might just know enough about their system that they have to pick KNX, but they would not know what to choose here. And maybe the target audience for KNX is the more technical kind of people who got trapped in KNX by accident, but I understand that the current question isn't that user friendly for the non-technical people

Reply:
The non-technical user should use "Automatic". Should I add a note to `"Automatic" will use the first free tunnel endpoint.`? Like
> "Automatic" selects any available free tunnel endpoint for you. If you're unsure, this is the best option.

For the standard installation of non technical users, "Automatic" in the first flow step will do, and this new step will never show. But for installations using encrypted communication (newer devices) - this is option is critical for successful bus communication. So imho hiding in the OptionsFlow would be bad UX too. 
Apart from that, I am planning to remove the OptionsFlow in favour of https://github.com/home-assistant/core/pull/96855 and move the reconfiguration of connection settings to a reconfigure-flow - moving the 2-3 options we have additionally in options flow to the custom panel.

Reply:
<img width="616" alt="Bildschirmfoto 2024-12-20 um 16 12 17" src="https://github.com/user-attachments/assets/5cf7d272-eb4b-4dce-b31f-4bf8c6cc3499" />